components/packages/index.rst

```diff
@@ -0,0 +1,259 @@
+      remote_package:
+        url: https://github.com/esphome/non-existant-repo
+        ref: main # optional
+        files: [file1.yml, file2.yml]
+        refresh: 1d # optional
+
+      # A single file can be expressed using `file` or `files` as a string
+      remote_package_two:
+        url: https://github.com/esphome/non-existant-repo
+        file: file1.yml # cannot be combined with `files`
+        # files: file1.yml
+
+      # shorthand form github://username/repository/[folder/]file-path.yml[@branch-or-tag]
+      remote_package_three: github://esphome/non-existant-repo/file1.yml@main
+
+Packages as Templates
+---------------------
+
+Since packages are incorporated using the ``!include`` system, variables can be provided to them. This means that
+packages can be used as *templates*, allowing complex or repetitive configurations to be stored in a package file
+and then incorporated into the configuration more than once.
+
+Packages may also contain a ``defaults`` block which provides subsitutions for variables not provided by the
+``!include`` block.
+
+As an example, if the configuration needed to support three garage doors using the ``gpio`` switch platform and the
+``time_based`` cover platform, it could be constructed like this:
+
+.. code-block:: yaml
+
+    # In config.yaml
+    packages:
+      left_garage_door: !include
+        file: garage-door.yaml
+        vars:
+          door_name: Left
+          door_location: left
+          open_switch_gpio: 25
+          close_switch_gpio: 26
+      middle_garage_door: !include
+        file: garage-door.yaml
+        vars:
+          door_name: Middle
+          door_location: middle
+          open_switch_gpio: 27
+          close_switch_gpio: 29
+      right_garage_door: !include
+        file: garage-door.yaml
+        vars:
+          door_name: Right
+          door_location: right
+          open_switch_gpio: 15
+          close_switch_gpio: 18
+          open_duration: "1min"
+          close_duration: "50s"
+
+
+.. code-block:: yaml
+
+    # In garage-door.yaml
+    defaults:
+      open_duration: "2.1min"
+      close_duration: "2min"
+
+    switch:
+      - id: open_${door_location}_door_switch
+        name: ${door_name} Garage Door Open Switch
+        platform: gpio
+        pin: ${open_switch_gpio}
+
+      - id: close_${door_location}_door_switch
+        name: ${door_name} Garage Door Close Switch
+        platform: gpio
+        pin: ${close_switch_gpio}
+
+    cover:
+      - platform: time_based
+        name: ${door_name} Garage Door
+
+        open_action:
+          - switch.turn_on: open_${door_location}_door_switch
+        open_duration: ${open_duration}
+
+        close_action:
+          - switch.turn_on: close_${door_location}_door_switch
+        close_duration: ${close_duration}
+
+        stop_action:
+          - switch.turn_off: open_${door_location}_door_switch
+          - switch.turn_off: close_${door_location}_door_switch
```

Comment:
Can we simplify these examples down to a bare minimum, just maybe a single entity with a single substitution and single default